Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TeamsController.cs

```diff
@@ -1075,6 +1079,118 @@ private async Task<List<ShiftsIntegResponse>> ProcessSwapShiftRequestApprovalAsy
+            var updateProps = new Dictionary<string, string>();
+            var swapRequest = this.GetRequest<SwapRequest>(jsonModel, "/swapRequests/", approved);
+            var postedShifts = jsonModel.Requests.Where(x => x.Url.Contains("/shifts/", StringComparison.InvariantCulture) && x.Method == "POST").ToList();
+
+            var offeredShiftMap = await this.shiftMappingEntityProvider.GetShiftMappingEntityByRowKeyAsync(swapRequest.SenderShiftId).ConfigureAwait(false);
+            var requestedShiftMap = await this.shiftMappingEntityProvider.GetShiftMappingEntityByRowKeyAsync(swapRequest.RecipientShiftId).ConfigureAwait(false);
+            var success = false;
+
+            try
+            {
+                var swapShiftRequestMapping = await this.swapShiftMappingEntityProvider.
+                    GetMapping(offeredShiftMap.RowKey, requestedShiftMap.RowKey).ConfigureAwait(false);
+                var kronosReqId = swapShiftRequestMapping.KronosReqId;
+                var kronosRequestingUserId = swapShiftRequestMapping.RequestorKronosPersonNumber;
+                var kronosRequestedUserId = swapShiftRequestMapping.RequestedKronosPersonNumber;
+
+                updateProps.Add("KronosRequestingPersonNumber", kronosRequestingUserId);
+                updateProps.Add("SwapShiftRequestID", swapRequest.Id);
+                updateProps.Add("KronosOpenShiftRequestId", kronosReqId);
+
+                if (!approved)
+                {
+                    // Deny in Kronos, Update mapping for Teams.
+                    success = await this.swapShiftController.ApproveSwapShiftInKronos(kronosReqId, kronosRequestingUserId, swapShiftRequestMapping, approved).ConfigureAwait(false);
+                    if (!success)
+                    {
+                        responseModelList.Add(GenerateResponse(swapRequest.Id, HttpStatusCode.BadRequest, null, null));
+                        return responseModelList;
+                    }
+
+                    swapShiftRequestMapping.ShiftsStatus = ApiConstants.Refused;
+                    await this.swapShiftMappingEntityProvider.AddOrUpdateSwapShiftMappingAsync(swapShiftRequestMapping).ConfigureAwait(false);
+                    responseModelList.Add(GenerateResponse(swapRequest.Id, HttpStatusCode.OK, null, null));
+                    return responseModelList;
+                }
+
+                this.telemetryClient.TrackTrace($"Process approval of {swapRequest.Id}", updateProps);
+
+                // approve in kronos
+                success = await this.swapShiftController.ApproveSwapShiftInKronos(kronosReqId, kronosRequestingUserId, swapShiftRequestMapping, approved).ConfigureAwait(false);
+                updateProps.Add("SuccessfullyApprovedInKronos", $"{success}");
+
+                if (success)
+                {
+                    responseModelList.Add(GenerateResponse(swapRequest.Id, HttpStatusCode.OK, null, null));
```

Comment:
I think we should generate the OK response after all the updates in table just to be sure it all worked okay.